src/components/Input.jsx

```diff
@@ -68,7 +68,10 @@ const Input = forwardRef(
 
     const handleOnBlur = e => {
       if (!disableTrimOnBlur && typeof value === "string") {
-        e.target.value = value.trim();
+        const trimmedValue = value.trim();
+        if (value === trimmedValue) return;
```

Comment:
We can't return here, it would prevent the custom `onBlur` handler from triggering. We can conditionally call `handleChange` instead.